src/main/java/hudson/plugins/git/UserMergeOptions.java

```diff
@@ -89,13 +99,23 @@ public String getRef() {
         return MergeCommand.Strategy.DEFAULT;
     }
 
+    @DataBoundSetter
+    public void setMergeStrategy(MergeCommand.Strategy mergeStrategy) {
+        this.mergeStrategy = mergeStrategy.toString(); // not .name() as you might expect! TODO in Turkey this will be e.g. recursıve
```

Comment:
Why not use the `MergeCommand.Strategy` type in the class too?
For the fast forward mode (`MergeCommand.GitPluginFastForwardMode`) that's already the case.

Reply:
I guess the problem is backward compatibility.
The persisted configuration would then use the enum names (uppercase) instead of the enum String representation (lowercase).

Reply:
Yes the concern was to maintain compatibility for existing serialized objects.